client/settings/check-starcraft-path.js

```diff
@@ -42,29 +53,37 @@ async function checkHash(path, validHashes) {
 // be checked for other copies. If downgradePath contains files that match the correct hashes, this
 // will be counted as having the correct version, but `downgradePath` will be true.
 export async function checkStarcraftPath(dirPath, downgradePath) {
-  const requiredFiles = ['starcraft.exe', 'storm.dll', 'local.dll', 'stardat.mpq', 'broodat.mpq']
+  const requiredFiles = ['starcraft.exe', 'storm.dll', 'stardat.mpq', 'broodat.mpq']
 
   try {
     await Promise.all(requiredFiles.map(f => accessAsync(path.join(dirPath, f), fs.constants.R_OK)))
   } catch (err) {
     return { path: false, version: false, downgradePath: false }
   }
 
+  // Due to 1.19 version moving local.dll to a separate folder, we need to handle it separately
+  let localDllValid = await checkFileExists(path.join(dirPath, 'local.dll'))
+  const matches = await globAsync(`${dirPath}/locales/*/local.dll`)
+  if (!localDllValid && matches.length < 1) {
+    return { path: false, version: false, downgradePath: false }
+  }
+
   let [starcraftValid, stormValid] = await Promise.all([
     checkHash(path.join(dirPath, 'starcraft.exe'), EXE_HASHES_1161),
     checkHash(path.join(dirPath, 'storm.dll'), STORM_HASHES_1161),
   ])
 
-  if (starcraftValid && stormValid) {
+  if (starcraftValid && stormValid && localDllValid) {
     return { path: true, version: true, downgradePath: false }
   }
 
+  localDllValid = await checkFileExists(path.join(downgradePath, 'local.dll'))
```

Comment:
This makes this a 2 step process instead of executing in parallel, you need to put this in the Promise.all call so they all get kicked off at once